### PR TITLE
100MB boot partition; Smaller img size

### PR DESCRIPTION
**Pull request by dominick-han**
100MB boot partition so we can do kernel upgrades easily
Auto resize image so we won't have 7GB image for a barebone linux

### CODE REVIEW

File: pieman.sh [hunk 1]
```diff
@@ -157,13 +157,13 @@ run_scripts "bootstrap"
 umount_required_filesystems
 
 info "creating image"
-dd if=/dev/zero of=${IMAGE} bs=1024 seek=$[ 1024 * 1024 * 7 ] count=1
+dd if=/dev/zero of=${IMAGE} bs=1024 seek=$[ `sudo du -s ${R}/ | cut -f1` + 150 * 1024 ] count=1
```
Comment (eugulixes) [inline pieman.sh]: It's redundant to use `sudo` in this case, because Pieman must be run either as root or using `sudo`.

**Comment by eugulixes** [comment]
@dominick-han, thank you very for much for your contribution. Could you take into account my comment and split the changes into two commits?
* Use 100MB boot partition
* Use smaller img size

The point is that the changes consist of two different tasks and it's better to keep the history clear.

**Comment by dominick-han** [comment]
@eugulixes Sorry! I usually do have the habit of splitting it into several commits, but these two depends on each other
Guess I'll change the boot partition size first then the image size :)

**Comment by eugulixes** [comment]
@dominick-han, great, thank you. I guess we have to discuss partition alignment because the topic is closely connected with you changes (see my comments). Then, I will be ready to merge your branch.

**Comment by dominick-han** [comment]
Updated to start at 4MB (official Raspbian image starts at 4MB)
I believe there are no reasons to leave a 4MB gap between the partitions, (there is a 485.5KB gap in the Raspbian image tho), but let me know if leaving gaps between partitions is actually a thing :)

File: pieman.sh [hunk 2]
```diff
@@ -157,13 +157,13 @@ run_scripts "bootstrap"
...
-dd if=/dev/zero of=${IMAGE} bs=1024 seek=$[ 1024 * 1024 * 7 ] count=1
+dd if=/dev/zero of=${IMAGE} bs=1024 seek=$[ `du -s ${R}/ | cut -f1` + 150 * 1024 ] count=1
 
 parted ${IMAGE} mktable msdos
 
-parted ${IMAGE} mkpart p fat32 4MiB 54MiB
+parted ${IMAGE} mkpart p fat32 4MiB 104MiB
 
-parted -s ${IMAGE} -- mkpart primary ext2 58MiB -1s
+parted -s ${IMAGE} -- mkpart primary ext2 104MiB -1s
```
Comment (eugulixes) [inline pieman.sh]: Let's align the second partition on 4MiB boundary too: 
`parted -s ${IMAGE} -- mkpart primary ext2 108MiB -1s`

Comment (dominick-han) [reply pieman.sh]: 104MB is aligned tho, do we have to leave a gap?

**Comment by eugulixes** [comment]
@dominick-han 

> I believe there are no reasons to leave a 4MB gap between the partitions, (there is a 485.5KB gap in the Raspbian image tho), but let me know if leaving gaps between partitions is actually a thing :)

The point is that we have to make some changes keeping the previous behaviour. The current implementation of Pieman aligns partitions on 4MiB boundaries. I don't mind removing the gap between partitions, but it should be a different PR where we can discuss pros and cons=) I like the changes you made and I really want to see them in `master`.

**Comment by dominick-han** [comment]
Completed :)

**Comment by eugulixes** [comment]
@dominick-han, thank you!

**Pull request merged**